src/sql/src/plan/statement.rs

```diff
@@ -2408,7 +2345,15 @@ impl<'a> StatementContext<'a> {
         Ok(())
     }
 
-    pub fn unwrap_param_types(self) -> BTreeMap<usize, ScalarType> {
-        Rc::try_unwrap(self.param_types).unwrap().into_inner()
+    pub fn finalize_param_types(self) -> Result<Vec<ScalarType>, anyhow::Error> {
+        let param_types = Rc::try_unwrap(self.param_types).unwrap().into_inner();
+        let mut out = vec![];
+        for (i, (n, typ)) in param_types.into_iter().enumerate() {
```

Comment:
Does this work because there won't be a `(n, typ)` tuple for something that failed to convert to a `ScalarType`?

Reply:
Yep exactly. We want to ensure the user specified a contiguous run of parameters starting for 1 so we just look for “holes” and bail out if we find any.